### PR TITLE
[#181853920] Mailer

### PR DESCRIPTION
**Pull request by AnnaAleynik**
Pivotal Tracker [#181853920](https://www.pivotaltracker.com/story/show/181853920)
Add mailer

### CODE REVIEW

File: spec/interactors/recipes/create_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    context "when valid params" do
+      let(:title) { "Cream Soup" }
+
+      it "creates recipe" do
+        interactor.run
+        expect(context).to be_success
+
+        recipe.reload
+
+        expect(recipe.title).to eq("Cream Soup")
```
Comment (GustAvGust) [inline spec/interactors/recipes/create_spec.rb:32]: `expect(recipe).to have_attributes(title: "Cream Soup", content: "Content")`

File: app/controllers/users/recipes_controller.rb [hunk 2]
```diff
@@ -28,6 +30,14 @@ def destroy
 
     private
 
+    def create_recipe
+      @create_recipe ||= ::CreateRecipe.call!(recipe: recipe, recipe_params: recipe_params, author: current_user)
```
Comment (GustAvGust) [inline app/controllers/users/recipes_controller.rb:34]: `::CreateRecipe.call!(recipe: recipe)`

File: spec/interactors/recipes/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    let(:initial_context) do
+      {
+        recipe: recipe,
+        author: user
+      }
+    end
+
+    let(:user) { create :user }
+    let(:recipe) { build :recipe, recipe_params }
```
Comment (GustAvGust) [inline spec/interactors/recipes/create_spec.rb:15]: `let(:recipe) { build, title: title, content: "Content" }` and we can remove `recipe_params` after that

File: config/initializers/letter_opener.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    # To render only the message body, without any metadata or extra containers or styling.
+    # Default value is `:default` that renders styled message with showing useful metadata.
+    config.message_template = :default
+
+    # To change default file URI scheme you can provide `file_uri_scheme` config.
+    # It might be useful when you use WSL (Windows Subsystem for Linux) and default
+    # scheme doesn't work for you.
+    # Default value is blank
+    config.file_uri_scheme = "file://///wsl$/Ubuntu-18.04"
```
Comment (GustAvGust) [inline config/initializers/letter_opener.rb:15]: try to remove this line and check does letter opener logic still works?

File: spec/interactors/create_recipe_spec.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    let(:expected_interactors) do
+      [
+        Recipes::Create
+      ]
+    end
+
+    it { is_expected.to eq(expected_interactors) }
+  end
+
```
Comment (GustAvGust) [inline spec/interactors/create_recipe_spec.rb:15]: `add describe ".call"`

File: spec/interactors/recipes/create_spec.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        expect(recipe).to have_attributes(title: "Cream Soup", content: "Content", author: user)
+      end
+    end
+
+    context "with invalid params" do
+      let(:title) { "N" }
+      let(:error) { "Title is too short (minimum is 3 characters)" }
+
+      it "raise failure" do
+        interactor.run
```
Comment (GustAvGust) [inline spec/interactors/recipes/create_spec.rb:34]: `expect { interactor.run }.not_to change(Recipe, :count) `

**Review by GustAvGust — APPROVED** [review]

File: config/initializers/letter_opener.rb (hunk 4 above)
Comment (AnnaAleynik) [reply config/initializers/letter_opener.rb]: This line is redundant!

**Pull request merged**